### PR TITLE
CLN: ci code checks

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by WillAyd — CHANGES_REQUESTED** [review]
Do these changes solve an issue and/or are you sure they work across platforms / shells? I’m somewhat hesitant on these

**Comment by MomIsBestFriend** [comment]
> Do these changes solve an issue

No, it will only solve my obnoxious OCD.

> are you sure they work across platforms / shells?

The thing is, I didn't write anything new, I just saw for example that on one line the ```include``` syntax is used like this:

```
--include "*.pyx" --include "*.py"
```

And on another line, it was used like:

```
--include=*.{py,pyx}
```

Because they're doing the exact same thing, I decided to pick one syntax (The second one just seems more compact), and apply it to every line that uses the ```--include``` flag.

File: ci/code_checks.sh [hunk 1]
```diff
@@ -121,82 +121,82 @@ if [[ -z "$CHECK" || "$CHECK" == "patterns" ]]; then
     # Check for imports from pandas.core.common instead of `import pandas.core.common as com`
     # Check for imports from collections.abc instead of `from collections import abc`
     MSG='Check for non-standard imports' ; echo $MSG
-    invgrep -R --include="*.py*" -E "from pandas.core.common import" pandas
-    invgrep -R --include="*.py*" -E "from pandas.core import common" pandas
-    invgrep -R --include="*.py*" -E "from collections.abc import" pandas
-    invgrep -R --include="*.py*" -E "from numpy import nan" pandas
+    invgrep -R --include=*.py* -E "from pandas.core.common import" pandas
+    invgrep -R --include=*.py* -E "from pandas.core import common" pandas
```
Comment (jreback) [inline ci/code_checks.sh]: are you sure this works? this generally needs to be quoted

**Review by jreback — CHANGES_REQUESTED** [review]

File: ci/code_checks.sh (hunk 1 above)
Comment (MomIsBestFriend) [reply ci/code_checks.sh]: I don't mind reverting.
I tested it on my machine.
```
Grub version = grep (GNU grep) 3.3

Bash version = GNU bash, version 5.0.11(1)-release (x86_64-pc-linux-gnu)
```

Plus, we have a few variations of the ```--include```/```--exclude```  already in the CI.
For example:

https://github.com/pandas-dev/pandas/blob/facd756fb0e6a2d9a8ac934ae4295c40a0c30102/ci/code_checks.sh#L203

https://github.com/pandas-dev/pandas/blob/facd756fb0e6a2d9a8ac934ae4295c40a0c30102/ci/code_checks.sh#L208

But if it's too risky, I'll just change everything to the ```safe``` syntax.

Comment (jreback) [reply ci/code_checks.sh]: this doesn’t work in macosx and likely not on windows (though not sure how much of this works on windows)

Comment (MomIsBestFriend) [reply ci/code_checks.sh]: Ok, so revert to quotes?
And if the answer is 'yes', double or single quotes?

Comment (jreback) [reply ci/code_checks.sh]: yes quotes; single quotes i think are appropriate here (but likely double do the same)

Comment (MomIsBestFriend) [reply ci/code_checks.sh]: @jreback, the problem now is that it doesn't work:

```invgrep -R --include='*.{py,pyx}' 'self' pandas```

or

```invgrep -R --include="*.{py,pyx}" 'self' pandas```

(not even if I switch the quotes on the regex and the wanted string, tried all four combinations)

should I just "exand" each statement?, meaning making every ```--include=*.{py,pyx}``` to ```--include='*.py' --include='*.pyx'```

Comment (jreback) [reply ci/code_checks.sh]: @MomIsBestFriend right I am not sure the expansion syntax is generally used on linux, ok with just making it a substion though, e.g.
```INCLUDES=--include='*.py'......``` and then using INCLUDES (and EXCLUDES) might be simpler

File: ci/code_checks.sh [hunk 2]
```diff
@@ -121,86 +121,86 @@ if [[ -z "$CHECK" || "$CHECK" == "patterns" ]]; then
     # Check for imports from pandas.core.common instead of `import pandas.core.common as com`
     # Check for imports from collections.abc instead of `from collections import abc`
     MSG='Check for non-standard imports' ; echo $MSG
-    invgrep -R --include="*.py*" -E "from pandas.core.common import" pandas
-    invgrep -R --include="*.py*" -E "from pandas.core import common" pandas
-    invgrep -R --include="*.py*" -E "from collections.abc import" pandas
-    invgrep -R --include="*.py*" -E "from numpy import nan" pandas
+    invgrep -R --include='*.py*' -E "from pandas.core.common import" pandas
```
Comment (jorisvandenbossche) [inline ci/code_checks.sh]: Why this change from double to single backticks? 
We shouldn't be changing things just for the sake of changing lines ..

Comment (MomIsBestFriend) [reply ci/code_checks.sh]: > We shouldn't be changing things just for the sake of changing lines ..

Ok, I understand
